File: code/classes/declaration.erl
% Erlang doesn't have classes in the traditional sense
% But it has records and modules that serve similar purposes
-module(person).
-export([new/2, greet/1]).

-record(person, {name, age}).

new(Name, Age) -&gt;
    #person{name=Name, age=Age}.

greet(#person{name=Name, age=Age}) -&gt;
    io_lib:format("Hello, I'm ~s and I'm ~p years old", [Name, Age]).

File: code/classes/declaration.ex
# Elixir doesn't have classes in the traditional sense
# But it has structs and modules that serve similar purposes
defmodule Person do
  defstruct name: nil, age: nil
  
  def new(name, age) do
    %Person{name: name, age: age}
  end
  
  def greet(person) do
    "Hello, I'm #{person.name} and I'm #{person.age} years old"
  end
end

person = Person.new("John", 30)
IO.puts Person.greet(person)